src/devices/Common/Iot/Device/Graphics/BdfFont.cs

```diff
@@ -0,0 +1,266 @@
+                    ReadNextDecimalNumber(ref span) != Height ||
+                    ReadNextDecimalNumber(ref span) != XDisplacement ||
+                    ReadNextDecimalNumber(ref span) != YDisplacement )
+                {
+                    throw new NotSupportedException("We don't support fonts have BBX values different than FONTBOUNDINGBOX values.");
+                }
+
+                span = sr.ReadLine().AsSpan().Trim();
+                if (span.CompareTo(s_bitmap.AsSpan(), StringComparison.Ordinal) != 0)
+                {
+                    throw new InvalidDataException("The font data is not well formed. expected BITMAP tag.");
+                }
+
+                span = sr.ReadLine().AsSpan().Trim();
+                int heightData = 0;
+                while (heightData < Height)
+                {
+                    if (span.Length > 0)
+                    {
+                        int data = ReadNextHexaDecimalNumber(ref span);
+                        GlyphUshortData[index] = (ushort)data;
+
+                        heightData++;
+                        index++;
+                    }
+                    else
+                    {
+                        span = sr.ReadLine().AsSpan().Trim();
+                    }
+                }
+
+                span = sr.ReadLine().AsSpan().Trim();
+                if (!span.StartsWith(s_endChar.AsSpan(), StringComparison.Ordinal))
+                {
+                    throw new InvalidDataException("The font data is not well formed. expected ENDCHAR tag in the beginning of glyoh data.");
+                }
+            }
+        }
+
+        private static int ReadNextDecimalNumber(ref ReadOnlySpan<char> span)
+        {
+            span = span.Trim();
+
+            int sign = 1;
+            if (span.Length > 0 && span[0] == '-')
+            {
+                sign = -1;
+                span = span.Slice(1);
+            }
+
+            int number = 0;
+            while (span.Length > 0 && ((uint) (span[0] - '0')) <= 9)
+            {
+                number = number * 10 + (span[0] - '0');
+                span = span.Slice(1);
+            }
+
+            return number * sign;
+        }
+
+        private static int ReadNextHexaDecimalNumber(ref ReadOnlySpan<char> span)
+        {
+            span = span.Trim();
+
+            int number = 0;
+            while (span.Length > 0)
+            {
+                if ((uint)(span[0] - '0') <= 9)
+                {
+                    number = number * 16 + (span[0] - '0');
+                    span = span.Slice(1);
+                    continue;
+                }
+                else if ((uint) (Char.ToLowerInvariant(span[0]) - 'a') <= ((uint) ('f' - 'a')))
+                {
+                    number = number * 16 + (Char.ToLowerInvariant(span[0]) - 'a') + 10;
+                    span = span.Slice(1);
+                    continue;
+                }
```

Comment:
do we care about uppercase hex?

Reply:
I am already doing toLower. so we support upper casing hex.

Reply:
right, missed it :)